code/modules/modular_computers/hardware/hard_drive.dm

```diff
@@ -174,3 +174,16 @@
 			var/datum/computer_file/data/F = SSdocs.create_file(file)
 			store_file(F)
 	. = ..()
+
+/obj/item/computer_hardware/hard_drive/proc/reset_drive()
+	for(var/datum/computer_file/F in stored_files)
+		remove_file(F)
+	install_default_programs()
+	store_file(new /datum/computer_file/program/ntnetdownload(src))
```

Comment:
The client and hardware manager applications are usually part of the default programs.
This allows to determine if the device is a company (no software downloader) or a private device.
Remove the manual addition of the downloader here.